looker/data_source_looker_role.go

```diff
@@ -0,0 +1,160 @@
+
+	sdk "github.com/looker-open-source/sdk-codegen/go/sdk/v4"
+)
+
+func datasourceRole() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceRoleRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+				Description:  "The name of the role. This field is case sensitive. See the documentation on looker [roles](https://docs.looker.com/admin-options/settings/roles) and [default roles](https://docs.looker.com/admin-options/settings/roles#default_roles) generated by Looker.",
+			},
+			"id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ExactlyOneOf: []string{"name", "id"},
+				Description:  "The id of the role",
+			},
+			"permission_set": {
+				Type: schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The name of the permission set",
+						},
+						"id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The id of the permission set",
+						},
+						"permissions": {
+							Type: schema.TypeSet,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Computed:    true,
+							Description: "An unordered list of models within the permission set",
+						},
+					},
+				},
+				Computed:    true,
+				Description: "The permission set binded to the looker role. This `permission_set` attribute holds additional attributes about the permission set.",
+			},
+			"model_set": {
+				Type: schema.TypeSet,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The name of the model set. This field is case sensitive",
+						},
+						"id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The id of the model set",
+						},
+						"models": {
+							Type: schema.TypeSet,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Computed:    true,
+							Description: "An unordered list of models within the model set.",
+						},
+					},
+				},
+				Computed:    true,
+				Description: "The model set binded to the looker role. This `model_set` attribute holds additional attributes about the model set",
+			},
+		},
+	}
+}
+
+func datasourceRoleRead(ctx context.Context, d *schema.ResourceData, c interface{}) diag.Diagnostics {
+	api := c.(*sdk.LookerSDK)
+
+	// exactly one of these variables will be nil - this is enforced by the data source schema
+	name := conv.PString(d.Get("name").(string))
+	id := conv.PString(d.Get("id").(string))
+
+	roles, roleErr := api.SearchRoles(sdk.RequestSearchRoles{}, nil)
```

Comment:
Should we have `Id` and `Name` on the request struct?

Reply:
Yes! Definitely - thank you :)